tools/install_demo_configuration.sh

```diff
@@ -21,455 +16,15 @@ else
-ADMIN_HASH_LINE=$(grep -n 'admin:' "$INTERNAL_USERS_FILE" | cut -f1 -d:)
-
-awk -v hashed_admin_password="$HASHED_ADMIN_PASSWORD" '
-  /^ *hash: *"\$2a\$12\$VcCDgh2NDk07JGN0rjGbM.Ad41qVR\/YFJcgHp0UGns5JDymv..TOG"/ {
-    sub(/"\$2a\$12\$VcCDgh2NDk07JGN0rjGbM.Ad41qVR\/YFJcgHp0UGns5JDymv..TOG"/, "\"" hashed_admin_password "\"");
-  }
-  { print }
-' "$INTERNAL_USERS_FILE" > temp_file && mv temp_file "$INTERNAL_USERS_FILE"
-
-#network.host
-if $SUDO_CMD grep --quiet -i "^network.host" "$OPENSEARCH_CONF_FILE"; then
-	: #already present
-else
-	if [ "$cluster_mode" == 1 ]; then
-        echo "network.host: 0.0.0.0" | $SUDO_CMD tee -a "$OPENSEARCH_CONF_FILE" > /dev/null
-        echo "node.name: smoketestnode" | $SUDO_CMD tee -a "$OPENSEARCH_CONF_FILE" > /dev/null
-        echo "cluster.initial_cluster_manager_nodes: smoketestnode" | $SUDO_CMD tee -a "$OPENSEARCH_CONF_FILE" > /dev/null
-    fi
-fi
-
-if $SUDO_CMD grep --quiet -i "^node.max_local_storage_nodes" "$OPENSEARCH_CONF_FILE"; then
-	: #already present
-else
-    echo 'node.max_local_storage_nodes: 3' | $SUDO_CMD tee -a "$OPENSEARCH_CONF_FILE" > /dev/null
-fi
-
-
-
-echo "######## End OpenSearch Security Demo Configuration ########" | $SUDO_CMD tee -a "$OPENSEARCH_CONF_FILE" > /dev/null
-
-$SUDO_CMD chmod +x "$OPENSEARCH_PLUGINS_DIR/opensearch-security/tools/securityadmin.sh"
-
-OPENSEARCH_PLUGINS_DIR=`cd "$OPENSEARCH_PLUGINS_DIR" ; pwd`
-
-echo "### Success"
-echo "### Execute this script now on all your nodes and then start all nodes"
-#Generate securityadmin_demo.sh
-echo "#!/bin/bash" | $SUDO_CMD tee securityadmin_demo.sh > /dev/null 
-echo $SUDO_CMD \""$OPENSEARCH_PLUGINS_DIR/opensearch-security/tools/securityadmin.sh"\" -cd \""$OPENSEARCH_CONF_DIR/opensearch-security"\" -icl -key \""$OPENSEARCH_CONF_DIR/kirk-key.pem"\" -cert \""$OPENSEARCH_CONF_DIR/kirk.pem"\" -cacert \""$OPENSEARCH_CONF_DIR/root-ca.pem"\" -nhnv | $SUDO_CMD tee -a securityadmin_demo.sh > /dev/null
-$SUDO_CMD chmod +x securityadmin_demo.sh
+BIN_PATH="java"
 
-if [ "$initsecurity" == 0 ]; then
-	echo "### After the whole cluster is up execute: "
-	$SUDO_CMD cat securityadmin_demo.sh | tail -1
-	echo "### or run ./securityadmin_demo.sh"
-    echo "### After that you can also use the Security Plugin ConfigurationGUI"
+# now set the path to java: first OPENSEARCH_JAVA_HOME, then JAVA_HOME
+if [ ! -z "$OPENSEARCH_JAVA_HOME" ]; then
+    BIN_PATH="$OPENSEARCH_JAVA_HOME/bin/java"
+elif [ ! -z "$JAVA_HOME" ]; then
+    BIN_PATH="$JAVA_HOME/bin/java"
 else
-    echo "### OpenSearch Security will be automatically initialized."
-    echo "### If you like to change the runtime configuration "
-    echo "### change the files in ../../../config/opensearch-security and execute: "
-	$SUDO_CMD cat securityadmin_demo.sh | tail -1
-	echo "### or run ./securityadmin_demo.sh"
-	echo "### To use the Security Plugin ConfigurationGUI"
+    echo "WARNING: nor OPENSEARCH_JAVA_HOME nor JAVA_HOME is set, will use $(which $BIN_PATH)"
```

Comment:
This should just be clear and show `which java` instead of `which $BIN_PATH` as it is `java` anyway.
You should also fail the script if you reach this `else` branch and still could not find `java` with `command -v`.

Reply:
Use `if command -v java >/dev/null` to see if java even exist in this else branch, if not, fail.

Reply:
Would we run into problems with rpm/deb by failing script?

Reply:
I would rather let it fail hard if there is no `java` found in path at all.
Also when you reach that third branch it means no OS_JAVA_HOME or JAVA_HOME anyway.

On deb/rpm there will always be a OS_JAVA_HOME there, so it will always goes to the 1st branch.
If we failed on the third branch we know this is an issue with core not bundling jdk, not your script.

In my opinion that is a even better scenario.
Thanks.

Reply:
ahh, I see. I will update the script. Ty!